ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/security/security.component.ts

```diff
@@ -237,4 +255,49 @@ export class SecurityComponent implements OnInit {
     toggleAdvancedMode() {
         this.inAdvancedMode = !this.inAdvancedMode;
     }
+
+    /**
+     * On Change event read the content of the file and add it to the
+     * corresponding array or overwrite the value at the given index
+     * @param {Event} evt change event on file input
+     * @param {string} targetField used to determine which array to push data to
+     * @param {number} index FormArray index
+     */
+    addFileContent(evt: Event, targetField: string, index: number) {
+        const fr = new FileReader();
+        const fileList: FileList = evt.target['files'];
+        const filereaderOnloadFactory = (filename: string) => {
+            return () => {
+                let targetArray: any[];
+                if (targetField === 'tlsCas') {
+                    targetArray = this.tlsCaContents;
+                } else if (targetField === 'registryCas') {
+                    targetArray = this.registryCaContents;
+                }
+
+                if (targetArray[index]) {
+                    // overwrite if value already exists at this index
+                    targetArray[index] = {
+                        name: filename,
+                        content: fr.result
+                    };
+                } else {
+                    targetArray.push({
+                        name: filename,
+                        content: fr.result
+                    });
+                }
+            };
+        };
+
+        // since input is without the 'multiple' attribute we are sure that
+        // only one entry will be available under FileList
+        const fileInstance: File = fileList[0];
+        if (!fileInstance) {
+            console.error('FileReader failed to load the file!');
```

Comment:
We should probably output this as an actual UI error to the user if there is a failure.